src/java/org/apache/cassandra/db/composites/Composite.java

```diff
@@ -73,6 +73,8 @@ public static EOC from(int eoc)
 
     public ByteBuffer toByteBuffer();
 
+    public int serializedSize();
```

Comment:
I don't know the backcompat constraints -- if there may be arbitrary implementations of `Composite` that we're unaware of, we could provide an expensive default implementation which delegates to `toByteBuffer().remaining()`